src/hotspot/share/cds/classListParser.cpp

```diff
@@ -459,18 +456,19 @@ void ClassListParser::error(const char* msg, ...) {
 
 void ClassListParser::check_class_name(const char* class_name) {
   const char* err = nullptr;
-  if (strlen(class_name) > (size_t)Symbol::max_length()) {
+  size_t len = strlen(class_name);
+  if (len > (size_t)Symbol::max_length()) {
     err = "class name too long";
   } else {
-    int len = (int)strlen(class_name);
-    if (!UTF8::is_legal_utf8((const unsigned char*)class_name, len, /*version_leq_47*/false)) {
+    assert(Symbol::max_length() < INT_MAX && len < INT_MAX, "must be");
```

Comment:
The first half of the assert is redundant as Symbol_max_length is fixed at 64K and will never change.